### ISSUE TITLE
Issue when passing multiple constraints for an optional dependency

### ISSUE THREAD

**Issue by patrick91** [python-poetry/poetry#4872]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [ ] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: MacOS Latest
- **Poetry version**: 1.1.12
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: [](https://gist.github.com/patrick91/42daa33d2eef46fa22bd1476aadcb5b6)

## Issue

Hi there! We just found a bug when passing multiple constraints for an optional package (https://github.com/strawberry-graphql/strawberry/issues/1488)

Basically when doing something like this:

```
django = [
    {version = ">=2.2,<4", python = "<3.8", optional = true},
    {version = ">=2.2,<5", python = ">=3.8", optional = true}
]
```

the second optional is ignore leading to django being a non optional dependency.

Apparently this is cause by the break here: https://github.com/python-poetry/poetry-core/blob/af08f1ce720da467c9bf3d43eed3d9ebaf4ad7fb/src/poetry/core/factory.py#L191-L196

Not sure if the break is meant as an optimisation measure or it is meant for something else :)

I can send a PR to poetry core if that's ok, I opened the issue here as it might be useful to other people (if they find the same issue)

**Comment by dimbleby** [comment]
```
[tool.poetry.dependencies]
python = "^3.7"
django = [
    {version = ">=2.2,<4", python = "<3.8", optional = true},
    {version = ">=2.2,<5", python = ">=3.8", optional = true}
]

[tool.poetry.extras]
django = ["django"]
```
results in django being installed as an optional dependency today.

Seems fixed.

**Comment by ralbertazzi** [comment]
Same here. Closing as stale and fixed
